src/components/App.jsx

```diff
@@ -11,48 +11,76 @@ import ImportResourceTemplate from './editor/ImportResourceTemplate'
 import Browse from './editor/Browse'
 import Login from './Login'
 import { loadState } from '../localStorage'
+import CognitoUtils from '../CognitoUtils.js'
+import { AuthenticationDetails, CognitoUser, CognitoUserPool } from 'amazon-cognito-identity-js'
+
 
 const FourOhFour = () => <h1>404</h1>
 
 class App extends Component{
   constructor(props) {
     super(props)
+    //TODO: would be nice to explore serializing currentuser to local storage, and attempting to load back here, before doing
+    // CognitoUtils.cognitoUserPool().getCurrentUser() (doing that if load from local storage fails).  BUT, that's a little weird,
+    // because amplify SDK (for which CognitoUtils.cognitoUserPool().getCurrentUser() is a thin wrapper) already tries to store user
+    // in local storage.  it just doesn't store the sign-in session.  the nice thing about saving that is that the user's session
+    // could persist between page loads, which it doesn't, at present.  all you get from using the current user here is the username (as
+    // far as what's useful to us).
+    let currentUser = CognitoUtils.cognitoUserPool().getCurrentUser()
     this.state = {
       redirectToReferrer: false,
-      isMenuOpened: false
+      isMenuOpened: false,
+      currentUser: currentUser,
+      currentSession: currentUser ? currentUser.getSignInUserSession() : null
     }
+    this.onUserAuthentication = this.onUserAuthentication.bind(this)
+    this.onUserSignout = this.onUserSignout.bind(this)
+  }
+
+  onUserAuthentication(authAttemptResult) {
```

Comment:
If you use the arrow function syntax, you can avoid having to do `this.function = this.function.bind(this)` in the constructor for every function defined in the component context. The arrow syntax implicitly binds the function to its context:

```
onUserAuthentication = (authAttemptResult) => {
  if ...
  else...
}
```